llvm/test/Transforms/LICM/hoist-binop.ll

```diff
@@ -71,7 +115,31 @@ loop:
   br label %loop
 }
 
-; Fold ADD but don't copy NSW if one op has it.
+; Hoist MUL and drop NUW if only one op has it.
```

Comment:
I'd probably drop the one nuw/nsw cases, because if we don't fold with both, then we won't do it for one either.